client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/MlClientDocumentationIT.java

```diff
@@ -2802,6 +2786,298 @@ public void onFailure(Exception e) {
+            List<DataFrameAnalyticsStats> stats = response.getAnalyticsStats();
+            // end::get-data-frame-analytics-stats-response
+
+            assertThat(stats.size(), equalTo(1));
+        }
+        {
+            GetDataFrameAnalyticsStatsRequest request = new GetDataFrameAnalyticsStatsRequest("my-analytics-config");
+
+            // tag::get-data-frame-analytics-stats-execute-listener
+            ActionListener<GetDataFrameAnalyticsStatsResponse> listener = new ActionListener<>() {
+                @Override
+                public void onResponse(GetDataFrameAnalyticsStatsResponse response) {
+                    // <1>
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    // <2>
+                }
+            };
+            // end::get-data-frame-analytics-stats-execute-listener
+
+            // Replace the empty listener by a blocking listener in test
+            CountDownLatch latch = new CountDownLatch(1);
+            listener = new LatchedActionListener<>(listener, latch);
+
+            // tag::get-data-frame-analytics-stats-execute-async
+            client.machineLearning().getDataFrameAnalyticsStatsAsync(request, RequestOptions.DEFAULT, listener); // <1>
+            // end::get-data-frame-analytics-stats-execute-async
+
+            assertTrue(latch.await(30L, TimeUnit.SECONDS));
+        }
+    }
+
+    private static final DataFrameAnalyticsConfig DF_ANALYTICS_CONFIG =
+        DataFrameAnalyticsConfig.builder("my-analytics-config")
+            .setSource(DataFrameAnalyticsSource.builder()
+                .setIndex("put-test-source-index")
+                .build())
+            .setDest(DataFrameAnalyticsDest.builder()
+                .setIndex("put-test-dest-index")
+                .build())
+            .setAnalysis(OutlierDetection.createDefault())
+            .build();
+
+    public void testPutDataFrameAnalytics() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+        {
+            // tag::put-data-frame-analytics-query-config
+            QueryConfig queryConfig = new QueryConfig(new MatchAllQueryBuilder());
+            // end::put-data-frame-analytics-query-config
+
+            // tag::put-data-frame-analytics-source-config
+            DataFrameAnalyticsSource sourceConfig = DataFrameAnalyticsSource.builder()
+                .setIndex("put-test-source-index")
+                .setQueryConfig(queryConfig)
+                .build();
+            // end::put-data-frame-analytics-source-config
+
+            // tag::put-data-frame-analytics-dest-config
+            DataFrameAnalyticsDest destConfig = DataFrameAnalyticsDest.builder()
+                .setIndex("put-test-dest-index")
+                .build();
+            // end::put-data-frame-analytics-dest-config
+
+            // tag::put-data-frame-analytics-analysis
+            DataFrameAnalysis analysis = OutlierDetection.createDefault();
```

Comment:
Do we want to document the `OutlierDetection` builder methods here as well? There are zero java docs for the class. We need to have at least some documentation (preferably here AND some java docs on the methods) on the builder methods.

Reply:
To make this future-proof, I think we should refer from here to another section where we will list creation of the various analysis types.

Reply:
Added a section about analysis. Not sure though how detailed it should be.